clarified from part 1 to part 4 - After Benjamin's version (#125)
trilitech/tezos-developer-docs: src/pages/tutorials/build-an-nft-marketplace/part-2.md

Explanation:

- Search for the offer based on the `token_id` or return an error if it does not exist
- Check that the amount sent by the buyer is greater than the offer price. If it is ok, transfer the offer price to the seller and transfer the NFT to the buyer

> - Remove the offer as it has been executed

Reviewer: ```suggestion
- Remove the offer from the storage
```

### Step 4: Compile and deploy

> Smart contract implementation of this second training is finished, let's deploy to ghostnet.

Reviewer: ```suggestion
Now that the smart contract is updated with buy and sell features, let's deploy it to ghostnet.
```

> **Smart contract (backend) is implmented and deployed!**

Reviewer: ```suggestion
Now the updated smart contract is deployed and the marketplace can use it.
```

## NFT Marketplace frontend

This section guides you step-by-step in setting up an intuitive frontend.

### Step 1: Initialize the Typescript classes

> Generate Typescript classes and go to the frontend to run the server

Reviewer: ```suggestion
Generate Typescript classes and go to the frontend to run the server:
```

yarn dev
```

### Step 2: Sale page

> Edit Sale Page on `./src/OffersPage.tsx`

Reviewer: ```suggestion
Follow these steps to implement a page to show the bottles for sale in the `./src/OffersPage.tsx` file:
```

> 1. Add this code inside the file :

Reviewer: ```suggestion
1. Add this code inside the file:
```

                                }
                                InputProps={{
                                  endAdornment: (
                                    <InputAdornment position="end">
                                      <Button
                                        type="submit"
                                        aria-label="add to favorites"
                                      >
                                        <SellIcon /> Sell
                                      </Button>
                                    </InputAdornment>
                                  ),
                                }}
                              />
                            </span>
                          </form>
                        )}
                      </CardActions>
                    </Card>
                  ))}{" "}
              </ImageList>
            </Fragment>
          ) : (
            <Typography sx={{ py: "2em" }} variant="h4">
              Sorry, you don't own any bottles, buy or mint some first
            </Typography>
          )}
        </Paper>
      );
    }
    ```

>     Explanation :

Reviewer: ```suggestion
    Explanation:
```

>     - The template displays all owned NFTs. Only NFTs belonging to the logged user are selected
>     - For each NFT, there is a form to make an offer at a price
>     - If you do an offer, it calls the `sell` function and the smart contract entrypoint `nftContrat?.methods.sell(BigNumber(token_id) as nat,BigNumber(price * 1000000) as nat).send()`. Multiply the XTZ price by 10^6 because the smart contract manipulates mutez.

Reviewer: ```suggestion
    - The template displays the NFTs that the logged-in user owns
    - Each NFT has a form to make an offer at a price
    - When the user submits the offer, the maketplace calls the `sell` function and the matching smart contract entrypoint
```

> ### Let's play : Sell

Reviewer: ```suggestion
### Let's play: Sell
```

> - Connect with your wallet and choose **alice** account (or one of the administrators you set on the smart contract earlier). You are redirected to the Administration /mint page as there is no NFT minted yet

Reviewer: ```suggestion
- Connect with your wallet and choose the **alice** account (or one of the administrators you set on the smart contract earlier). You are redirected to the Administration /mint page because there is no NFT minted yet
```

> - Enter these values on the form for example :

Reviewer: ```suggestion
- Enter values on the form to create an NFT, for example:
```

- Click on **SELL** button
- Wait a bit for the confirmation, then after auto-refresh you have an offer for this NFT

> ### Step 3: Wine Catalogue page

Reviewer: ```suggestion
### Step 3: Wine catalogue page
```

> Edit the Wine Catalogue page on `./src/WineCataloguePage.tsx`

Reviewer: ```suggestion
Follow these steps to implement the wine catalogue page in the `./src/WineCataloguePage.tsx` file:
```

> 1. Add the following code inside the file

Reviewer: ```suggestion
1. Add the following code inside the file:
```

                              setUserBalance={setUserBalance}
                              wallet={wallet}
                            />
                          </Box>
                        ) : (
                          <form
                            style={{ width: "100%" }}
                            onSubmit={(values) => {
                              setSelectedOfferEntry([token_id, offer]);
                              formik.handleSubmit(values);
                            }}
                          >
                            <Button type="submit" aria-label="add to favorites">
                              <ShoppingCartIcon /> BUY
                            </Button>
                          </form>
                        )}
                      </CardActions>
                    </Card>
                  ))}
              </ImageList>
            </Fragment>
          ) : (
            <Typography sx={{ py: "2em" }} variant="h4">
              Sorry, there is not NFT to buy yet, you need to mint or sell bottles
              first
            </Typography>
          )}
        </Paper>
      );
    }
    ```

## Buy some wine!

> 

Reviewer: "Now you can see on **Trading** menu the **Wine catalogue** submenu, click on it." => "In the **Wine catalogue** submenu, click **Trading**."

- Click on **Sell bottle** sub menu
- You are now the owner of this bottle, you can resell it at your own price, etc ...

> 

Reviewer: "As you are connected with the default administrator you can see your own unique offer on the market" => "While you are connected with the administrator account you can see the offer in the marketplace page."

Reviewer: The list should be ordered steps instead of an unordered list.

## Summary

> You created an NFT collection marketplace from the Ligo library, now you can buy and sell NFTs at your own price.

Reviewer: Fix this comma splice.
